analytics-core/src/main/java/com/liferay/analytics/android/dao/UserDAO.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.analytics.android.dao
+
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.gson.reflect.TypeToken
+import com.liferay.analytics.android.model.IdentityContext
+import com.liferay.analytics.android.util.FileStorage
+import java.io.IOException
+import java.lang.reflect.Type
+
+/**
+ * @author Igor Matos
+ */
+internal class UserDAO(private var fileStorage: FileStorage) {
+
+	val userContexts: List<IdentityContext>
+		get() {
```

Comment:
I have seen this pattern a lot. Although this seems cleaner than having a method, I think it would be better to have a method here because the caller won't expect that accessing a variable will trigger an IO operation.

Reply:
Makes sense, @victorg1991! I will refactor that =)